homeassistant/components/combined_energy/sensor.py

```diff
@@ -0,0 +1,485 @@
+            translation_key="grid_meter_voltage_a",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+        ),
+        SensorEntityDescription(
+            key="voltage_b",
+            translation_key="grid_meter_voltage_b",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="voltage_c",
+            translation_key="grid_meter_voltage_c",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            entity_registry_enabled_default=False,
+        ),
+    ],
+    "GENERIC_CONSUMER": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+    "ENERGY_BALANCE": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+
+    api: CombinedEnergy = hass.data[DOMAIN][entry.entry_id][DATA_API_CLIENT]
+    installation: Installation = hass.data[DOMAIN][entry.entry_id][DATA_INSTALLATION]
+
+    # Initialise readings coordinator
+    readings = CombinedEnergyReadingsCoordinator(hass, api)
+    await readings.async_config_entry_first_refresh()
+
+    async_add_entities(_generate_sensors(installation, readings))
+
+
+def _generate_sensors(
+    installation: Installation,
+    readings: CombinedEnergyReadingsCoordinator,
+) -> Generator[CombinedEnergyReadingsSensor, None, None]:
+    """Generate sensor entities from installed devices."""
+
+    for device in installation.devices:
+        if descriptions := SENSOR_DESCRIPTIONS.get(device.device_type):
+            # Generate sensors from descriptions for the current device type
+            for description in descriptions:
+                if sensor_type := SENSOR_TYPE_MAP.get(
+                    description.device_class, GenericSensor
+                ):
+                    yield sensor_type(device, description, readings)
+
+
+class CombinedEnergyReadingsSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a Combined Energy API reading energy sensor."""
+
+    entity_description: SensorEntityDescription
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        device: Device,
+        description: SensorEntityDescription,
+        coordinator: CombinedEnergyReadingsCoordinator,
+    ) -> None:
+        """Initialise Readings Sensor."""
+        super().__init__(coordinator)
+
+        self.device_id = device.device_id
+        self.entity_description = description
+
+        identifier = (
+            f"install_{self.coordinator.api.installation_id}-device_{device.device_id}"
```

Comment:
Why are you extending the identifier by `install` and `device`?

Reply:
I have no guarantee that device_id is globally unique, I do however have a guarantee that a combination of installation_id and device id is.

Reply:
```suggestion
            f"{self.coordinator.api.installation_id}_{device.device_id}"
```

Reply:
Why did you only mark it as resolved without implementing the suggestion or leaving any comments?